File: app/src/main/java/com/bus/routes/module/ActivityModule.java
package com.bus.routes.module;

import android.app.Activity;
import android.content.Context;

import dagger.Module;
import dagger.Provides;

/**
 * Created by Gustavo on 16/06/18.
 */

@Module
public class ActivityModule {

    private Activity mActivity;

    public ActivityModule(Activity activity) {
        mActivity = activity;
    }

    @Provides
    @ActivityContext
    Context provideContext() {
        return mActivity;
    }

    @Provides
    Activity provideActivity() {
        return mActivity;
    }
}

File: app/src/main/java/com/bus/routes/module/ActivityModule.kt
package com.bus.routes.module

import android.app.Activity
import android.content.Context

import dagger.Module
import dagger.Provides

/**
 * Created by Gustavo on 16/06/18.
 */

@Module
class ActivityModule(private val mActivity: Activity) {

    @Provides
    @ActivityContext
    internal fun provideContext(): Context {
        return mActivity
    }

    @Provides
    internal fun provideActivity(): Activity {
        return mActivity
    }
}
